### ISSUE TITLE
selection the classes of observations using heatmaply in R

### ISSUE THREAD

**Issue by jasperDD** [talgalili/heatmaply#158]
The simple example

    library(heatmaply)
    heatmaply(mtcars, k_col = 2, k_row = 3) %>% layout(margin = list(l = 130, b = 40))
and we get this plot
[![plot][1]][1]
https://i.stack.imgur.com/hSRyk.jpg
We see that the observations were divided on 3 clusters

    1 from honda civic to ferrari dino
    2 from valiant to dodge challendger
    3 from chrysler imperial to maserati bora

Also we see that technical parameters(variables) were divided by two classes

    The first class **hp** and **disp**
    The second class another parameters

how can I output in dataframe the name(or number) of those observations that, on the one hand, they in the first cluster (cars of the first cluster),
and they  are located at the area of the first cluster of technical variables.
In other words, I identified this area on screen below

[![area][2]][2]
https://i.stack.imgur.com/vtrdu.png

Accordingly, the output must be made for observations of all clusters of cars in different clusters of technical variables.
It is easy to calculate that it will be 6 areas.

post on stackoverflow
https://stackoverflow.com/questions/48965115/selection-the-classes-of-observations-using-heatmaply-in-r

**Comment by Alanocallaghan** [comment]
Hi. thanks for the report. The package author Tal will likely be able to explain this properly, but very quickly, you could review the lines that deal with `k_col` and `k_row` ([here](https://github.com/talgalili/heatmaply/blob/master/R/heatmapr.R#L444)) and notice that it calls [`dendextend::color_branches`](https://www.rdocumentation.org/packages/dendextend/versions/1.6.0/topics/color_branches). 

As it stands, the groups cannot be extracted from heatmaply, but what you can do is to perform hierarchical clustering, apply labels using `color_branches` and then use that dendrogram in a call to `heatmaply`, eg `heatmaply(mtcars, Colv = mydend)`

**Comment by jasperDD** [comment]
Thank you:) can you show me 
i did so
library("pvclust")
Result<-pvclust(mtcars, method.hclust="average", method.dist="cor", quiet=TRUE)
heatmaply(mtcars, Colv = Result)
and there is no result=(

**Comment by Alanocallaghan** [comment]
That is because the result of `pvclust(...)` is of class `pvclust`, which is not supported. However this `pvclust` object contains a `hclust` object, which is supported. Try this:

```r
library("pvclust")
library("heatmaply")
Result<-pvclust(mtcars, method.hclust="average", method.dist="cor", quiet=TRUE)
heatmaply(mtcars, Colv = Result$hclust)
```

**Comment by jasperDD** [comment]
thank you, very well , but can save  the observation affiliation to the cluster in pvclust?

**Comment by Alanocallaghan** [comment]
I don't quite understand, could you be more specific?

**Comment by jasperDD** [comment]
for example, in the spss program, you can keep the observation belonging to the cluster. How to do it in pvclust?
like in this screen
https://imgur.com/a/7Bcv4

in this code
Result<-pvclust(mtcars, method.hclust="average", method.dist="cor", quiet=TRUE)
heatmaply(mtcars, k_col = 2, k_row = 3, Colv = Result$hclust)
i need to know I'm interested in how the observations on three clusters were distributed

**Comment by Alanocallaghan** [comment]
I see. I'm afraid I'm busy right now (and I was last night, hence the quick reply) but I'll try to post some example code later today. Thanks.

**Comment by jasperDD** [comment]
The biggest thank you

**Comment by Alanocallaghan** [comment]
Seems like something we could think about. We could add more functionality to the `heatmapr` class, along those lines

**Comment by jasperDD** [comment]
Thank you dear Friends. You helped me:)
